### ISSUE TITLE
MarkDuplicates should mark unmapped reads as duplicates when their mapped mates are marked.

### ISSUE THREAD

**Issue by yfarjoun** [broadinstitute/picard#1138]
## Bug Report
Currently (according to #1018) MarkDuplicates doesn't mark as duplicates the unmapped mates of mapped reads which are marked. This seems to be a bug. IMHO the correct behavior is that all reads from a given template should be consistent in their duplicate flag. It has little effect when reads are not removed, since most programs that ignore duplicates also ignore unmapped reads, but this is increasingly incorrect as programs make use of unmapped reads and would stand to benefit from the duplicate flag for these reads (@cwhelan ... Care to chime in here?)

While marking secondary and supplementary reads is only feasible (without another pass) in queryname-sorted input, there doesn't seem to be a technical limitation to marking the unmapped mates of reads. 

I've opened this issue prior to a PR since I want to hear public opinion on the matter before diving into the code and trying to fix this. If you have an opinion (either way) please voice it here so it can be considered as the issue is debated.

 
### Affected tool(s)
MarkDuplicates coordinate-sorted

### Affected version(s)
- [x] Latest public release version 
- [x] Latest development/master branch as of

**Comment by AlijahArcher** [comment]
Any update on this issue?  It appears to still be the case in Picard v2.21.9 (GATK v4.1.7)

**Comment by yfarjoun** [comment]
yeah, though note that if you mark duplicates on an input file that is query-name sorted, it does the right thing....so perhaps the fact that there's a work-around means that there's less pressure for this feature to be added.....

**Comment by AlijahArcher** [comment]
Yeah, I just noticed https://github.com/broadinstitute/picard/issues/451 that discussed this as well.  I'll have to think about what this workaround would look like in my pipeline
